### PR TITLE
OCM-7860 | fix: add detach commands for arbitrary policies

### PR DESCRIPTION
**Pull request by marcolan018**
[OCM-7860](https://issues.redhat.com//browse/OCM-7860) | fix: add detach commands for arbitrary policies in rosa delete roles commands
https://issues.redhat.com/browse/OCM-7860
Signed-off-by: marcolan018 <llan@redhat.com>

### CODE REVIEW

File: cmd/dlt/accountroles/cmd.go [hunk 1]
```diff
@@ -229,11 +229,11 @@ func deleteAccountRoles(r *rosa.Runtime, env string, prefix string, clusters []*
 		r.Reporter.Infof(fmt.Sprintf("Successfully deleted the %saccount roles", roleTypeString))
 	case interactive.ModeManual:
 		r.OCMClient.LogEvent("ROSADeleteAccountRoleModeManual", nil)
-		policyMap, err := r.AWSClient.GetAccountRolePolicies(finalRoleList, prefix)
+		policies, excludedPolicies, err := r.AWSClient.GetAccountRolePolicies(finalRoleList, prefix)
```
Comment (davidleerh) [inline cmd/dlt/accountroles/cmd.go:232]: Would it be better to align the variable names as what was done with delete operator roles?
https://github.com/openshift/rosa/pull/2037/files#diff-a7b239b718526b39104909fe028dfe41eaf6f7e6210f739e11b58dd45494ef11R254

WDYT? Would you kindly advise if there was a reason to differentiate naming used between 
excludedPolicies and arbitraryPolicies?

Comment (marcolan018) [reply cmd/dlt/accountroles/cmd.go]: @davidleerh 
thanks for pointing it out, that inconsistence is not desired. For the naming, in pkg, in the common function to get policies with tag filter, I use excludedPolicies for common scenarios, but in cmd of account/operator-role, I use arbitrarypolicies for specific usage
Please help to re-reivew the pr.

File: pkg/aws/policies.go [hunk 2]
```diff
@@ -1256,35 +1256,39 @@ func (c *awsClient) deletePolicyVersions(policyArn string) error {
...
+	tagFilter map[string]string) ([]PolicyDetail, []PolicyDetail, error) {
 	policies := []PolicyDetail{}
+	excludedPolicies := []PolicyDetail{}
 	attachedPoliciesOutput, err := c.iamClient.ListAttachedRolePolicies(
 		context.Background(),
 		&iam.ListAttachedRolePoliciesInput{RoleName: role},
 	)
 	if err != nil && !awserr.IsNoSuchEntityException(err) {
-		return policies, err
+		return policies, excludedPolicies, err
```
Comment (davidleerh) [inline pkg/aws/policies.go:1268]: should this also be updated to arbitraryPolicies for consistency?

File: pkg/aws/policies.go [hunk 3]
```diff
@@ -1256,35 +1256,39 @@ func (c *awsClient) deletePolicyVersions(policyArn string) error {
...
+		} else {
+			excludedPolicies = append(excludedPolicies, policyDetail)
 		}
 	}
 
 	rolePolicyOutput, err := c.iamClient.ListRolePolicies(context.Background(),
 		&iam.ListRolePoliciesInput{RoleName: role})
 	if err != nil && !awserr.IsNoSuchEntityException(err) {
-		return policies, err
+		return policies, excludedPolicies, err
```
Comment (davidleerh) [inline pkg/aws/policies.go:1291]: should this also be updated to arbitraryPolicies for consistency?

File: pkg/aws/policies.go [hunk 4]
```diff
@@ -1294,11 +1298,12 @@ func (c *awsClient) GetAttachedPolicyWithTags(role *string, tagFilter map[string
 		policies = append(policies, policyDetail)
 	}
 
-	return policies, nil
+	return policies, excludedPolicies, nil
```
Comment (davidleerh) [inline pkg/aws/policies.go:1301]: should this also be updated to arbitraryPolicies for consistency?

File: pkg/aws/policies.go [hunk 5]
```diff
@@ -1530,32 +1535,37 @@ func (c *awsClient) buildRoles(roleName string, accountID string) ([]Role, error
...
 
-func (c *awsClient) GetAccountRolePolicies(roles []string, prefix string) (map[string][]PolicyDetail, error) {
-	roleMap := make(map[string][]PolicyDetail)
+func (c *awsClient) GetAccountRolePolicies(roles []string, prefix string) (map[string][]PolicyDetail,
+	map[string][]PolicyDetail, error) {
+	rolePolicies := make(map[string][]PolicyDetail)
+	roleExcludedPolicies := make(map[string][]PolicyDetail)
 	for _, role := range roles {
-		policies, err := c.GetAttachedPolicyWithTags(aws.String(role), getAcctRolePolicyTags(prefix))
+		policies, excludedPolicies, err := c.GetAttachedPolicyWithTags(aws.String(role), getAcctRolePolicyTags(prefix))
```
Comment (davidleerh) [inline pkg/aws/policies.go:1543]: should this also be updated to arbitraryPolicies for consistency?

File: pkg/aws/policies.go [hunk 6]
```diff
@@ -1530,32 +1535,37 @@ func (c *awsClient) buildRoles(roleName string, accountID string) ([]Role, error
...
 		if err != nil && !awserr.IsNoSuchEntityException(err) {
-			return roleMap, err
+			return rolePolicies, roleExcludedPolicies, err
```
Comment (davidleerh) [inline pkg/aws/policies.go:1545]: should this also be updated to arbitraryPolicies for consistency?

File: pkg/aws/policies.go [hunk 7]
```diff
@@ -1530,32 +1535,37 @@ func (c *awsClient) buildRoles(roleName string, accountID string) ([]Role, error
...
-func (c *awsClient) GetOperatorRolePolicies(roles []string) (map[string][]string, error) {
-	roleMap := map[string][]string{}
+func (c *awsClient) GetOperatorRolePolicies(roles []string) (map[string][]string, map[string][]string, error) {
+	rolePolicies := map[string][]string{}
+	roleExcludedPolicies := map[string][]string{}
 	for _, role := range roles {
 		tagFilter, err := getOperatorRolePolicyTags(c.iamClient, role)
 		if err != nil {
-			return roleMap, err
+			return rolePolicies, roleExcludedPolicies, err
```
Comment (davidleerh) [inline pkg/aws/policies.go:1559]: should this also be updated to arbitraryPolicies for consistency?

File: pkg/aws/policies.go [hunk 8]
```diff
@@ -1530,32 +1535,37 @@ func (c *awsClient) buildRoles(roleName string, accountID string) ([]Role, error
...
 		}
-		policies, err := getAttachedPolicies(c.iamClient, role, tagFilter)
+		policies, excludedPolicies, err := getAttachedPolicies(c.iamClient, role, tagFilter)
```
Comment (davidleerh) [inline pkg/aws/policies.go:1561]: should this also be updated to arbitraryPolicies for consistency?

File: pkg/aws/policies.go [hunk 9]
```diff
@@ -1530,32 +1535,37 @@ func (c *awsClient) buildRoles(roleName string, accountID string) ([]Role, error
...
 		if err != nil {
-			return roleMap, err
+			return rolePolicies, roleExcludedPolicies, err
```
Comment (davidleerh) [inline pkg/aws/policies.go:1563]: should this also be updated to arbitraryPolicies for consistency?

**Comment by davidleerh** [comment]
After manual test~ working as expected. Thank you!

**Pull request merged**